game-app/game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -1811,11 +1779,7 @@ private void resetCanDieFromReachingMaxDamage() {
   @VisibleForTesting
   public void setIsSuicide(final Boolean s) {
```

Comment:
Hmm, that's a good point, I think this is a bug?

We should do `isSuicide = s;`.

The existing code doesn't handle `<option name="isSuicide" value="false"/>` correctly, because `isSuicide` is always set to true!

I have a downloadedMaps folder with all available TripleA maps and I searched through the XML files and didn't see any occurrences of the above, but I did see instances of:
`            <option name="unitProperty" value="isSuicide" count="false"/>`

i.e. setting the property to false via a trigger. This is on `total_world_war`. So I guess we need to check with that map whether fixing it has an unintended consequence or whether it will fix a bug that exists on that map due to this not working.

Reply:
Looking more at how this is used in `total_world_war` map, they do two triggers:]

```
       <attachment name="triggerAttachmentjapaneseAdvRock2" attachTo="Japan" javaClass="TriggerAttachment" type="player">
            <option name="trigger" value="conditionAttachmentjapaneseAdvancedRockets"/>
            <option name="unitType" value="japaneseAdvancedRocket"/>
            <option name="unitProperty" value="isAir" count="false"/>
            <option name="unitProperty" value="isStrategicBomber" count="false"/>
            <option name="unitProperty" value="isSuicide" count="false"/>
            <option name="unitProperty" value="isKamikaze" count="false"/>
            <option name="unitProperty" value="isLandTransportable" count="true"/>
            <option name="when" value="before:japanNonCombatMove"/>
        </attachment>
        <attachment name="triggerAttachmentjapaneseAdvRock3" attachTo="Japan" javaClass="TriggerAttachment" type="player">
            <option name="trigger" value="conditionAttachmentjapaneseAdvancedRockets"/>
            <option name="unitType" value="japaneseAdvancedRocket"/>
            <option name="unitProperty" value="isLandTransportable" count="false"/>
            <option name="unitProperty" value="isAir" count="true"/>
            <option name="unitProperty" value="isStrategicBomber" count="true"/>
            <option name="unitProperty" value="isSuicide" count="true"/>
            <option name="unitProperty" value="isKamikaze" count="true"/>
            <option name="when" value="after:japanNonCombatMove"/>
        </attachment>
```

So basically, before NonCombatMove, the rockets are made land transportable and after NonCombatMove, they're returned back to non-land transportable. (And the suicide property is also changed to false then to true.)

So I think fixing isSuicide=false to work correctly shouldn't cause issues for this map and it's the only one setting it to false. I'll go ahead and fix it.